diff --git a/crates/maf_typed/src/typescript.rs b/crates/maf_typed/src/typescript.rs
index 5b91e1e..ac7525a 100644
--- a/crates/maf_typed/src/typescript.rs
+++ b/crates/maf_typed/src/typescript.rs
@@ -51,6 +51,7 @@ impl TypeScriptCodegen {
 
         output.push_str(
             r#"export type MafApp = {
+  __isTyped: true;
   stores: MafStoresList;
   rpcs: MafRpcsList;
 };
diff --git a/packages/client/src/client.ts b/packages/client/src/client.ts
index 67e6e2e..3ed2a47 100644
--- a/packages/client/src/client.ts
+++ b/packages/client/src/client.ts
@@ -2,12 +2,23 @@ import Emittery from "emittery";
 import { Channel } from "./channel";
 import { Store, StoreOptions } from "./store";
 import { RxPacket, TxPacket } from "./packet";
-import type { MafTypes } from "./typed";
+import type { TypedMafClient } from "./typed";
 
 export interface MafClientOptions {
   server: MafServerOptions;
 }
 
+/**
+ * Options for connecting to a MAF server.
+ *
+ * This will be in the form of either:
+ * - `{ type: "dev" }` or `"dev"` (as a string literal) to connect to the
+ * default development server at `http://localhost:1147`.
+ * - `{ type: "dev", url: URL | string }` to connect to a development server
+ * hosted at a custom URL.
+ * - `{ type: "platform", url: URL | string, app: string }` to connect to MAF
+ * Platform at the specified URL for the specified app.
+ */
 export type MafServerOptions =
   | { type: "dev"; url?: string | URL }
   | {
@@ -19,6 +30,11 @@ export type MafServerOptions =
 
 export const DEFAULT_DEV_SERVER_URL = "http://localhost:1147";
 
+/**
+ * Events emitted by the {@link MafClient}
+ *
+ * @see MafClient
+ */
 export interface MafClientEvents {
   ready: SessionInfo;
   close: void;
@@ -255,6 +271,9 @@ export class MafUntypedBaseClient extends Emittery<MafClientEvents> {
   }
 }
 
+/**
+ * @see TypedMafClient
+ */
 export class MafClient extends MafUntypedBaseClient {
   constructor(options: MafClientOptions) {
     super(options);
diff --git a/packages/client/src/typed.ts b/packages/client/src/typed.ts
index a4b026c..a83c210 100644
--- a/packages/client/src/typed.ts
+++ b/packages/client/src/typed.ts
@@ -14,6 +14,10 @@ export interface RpcDefinition<P, R extends unknown | unknown[]> {
 
 interface DefaultMafTypes {
   generated: {
+    /**
+     * If `true`, this indicates that the generated types have been populated.
+     */
+    __isTyped: false;
     stores: Record<string, StoreDefinition<unknown>>;
     rpcs: Record<string, RpcDefinition<unknown, unknown>>;
   };
diff --git a/packages/react/examples/providers/src/app.tsx b/packages/react/examples/providers/src/app.tsx
index a0cf2d1..dd69c18 100644
--- a/packages/react/examples/providers/src/app.tsx
+++ b/packages/react/examples/providers/src/app.tsx
@@ -1,18 +1,19 @@
-import { useRpc, useStore } from "@usemaf/react";
+import { useStore, useMafClient } from "@usemaf/react";
 
 export const App: React.FC = () => {
-  const { data } = useStore<number>("count", 0);
-  const incr_mutation = useRpc<number>("increment_counter", 1);
+  const { data } = useStore("count", 0);
+  const maf = useMafClient();
 
   return (
     <>
-      <h1>lets fuckin go bro</h1>
-      <div className="card">
-        <button onClick={incr_mutation.mutateAsync}>count is {data}</button>
-      </div>
-      <p className="read-the-docs">
-        Click on the Ur Mom and Eat Farts logos to learn more
-      </p>
+      <p>The count is {data}</p>
+      <button
+        onClick={() => {
+          maf.rpc("increment_counter", 1);
+        }}
+      >
+        Click me to increment!
+      </button>
     </>
   );
 };
diff --git a/packages/react/examples/providers/src/index.css b/packages/react/examples/providers/src/index.css
deleted file mode 100644
index 6119ad9..0000000
--- a/packages/react/examples/providers/src/index.css
+++ /dev/null
@@ -1,68 +0,0 @@
-:root {
-  font-family: Inter, system-ui, Avenir, Helvetica, Arial, sans-serif;
-  line-height: 1.5;
-  font-weight: 400;
-
-  color-scheme: light dark;
-  color: rgba(255, 255, 255, 0.87);
-  background-color: #242424;
-
-  font-synthesis: none;
-  text-rendering: optimizeLegibility;
-  -webkit-font-smoothing: antialiased;
-  -moz-osx-font-smoothing: grayscale;
-}
-
-a {
-  font-weight: 500;
-  color: #646cff;
-  text-decoration: inherit;
-}
-a:hover {
-  color: #535bf2;
-}
-
-body {
-  margin: 0;
-  display: flex;
-  place-items: center;
-  min-width: 320px;
-  min-height: 100vh;
-}
-
-h1 {
-  font-size: 3.2em;
-  line-height: 1.1;
-}
-
-button {
-  border-radius: 8px;
-  border: 1px solid transparent;
-  padding: 0.6em 1.2em;
-  font-size: 1em;
-  font-weight: 500;
-  font-family: inherit;
-  background-color: #1a1a1a;
-  cursor: pointer;
-  transition: border-color 0.25s;
-}
-button:hover {
-  border-color: #646cff;
-}
-button:focus,
-button:focus-visible {
-  outline: 4px auto -webkit-focus-ring-color;
-}
-
-@media (prefers-color-scheme: light) {
-  :root {
-    color: #213547;
-    background-color: #ffffff;
-  }
-  a:hover {
-    color: #747bff;
-  }
-  button {
-    background-color: #f9f9f9;
-  }
-}
diff --git a/packages/react/examples/providers/src/main.tsx b/packages/react/examples/providers/src/main.tsx
index cf2d50f..27801c9 100644
--- a/packages/react/examples/providers/src/main.tsx
+++ b/packages/react/examples/providers/src/main.tsx
@@ -3,7 +3,6 @@ import { createRoot } from "react-dom/client";
 import { MafProvider, useStore } from "@usemaf/react";
 import { App } from "./app";
 import type { MafApp } from "./types";
-import "./index.css";
 
 declare module "@usemaf/client" {
   interface MafTypes {
diff --git a/packages/react/examples/providers/src/types.ts b/packages/react/examples/providers/src/types.ts
index f346e6d..685e110 100644
--- a/packages/react/examples/providers/src/types.ts
+++ b/packages/react/examples/providers/src/types.ts
@@ -18,6 +18,7 @@ export type MafRpcsList = {
 };
 
 export type MafApp = {
+  __isTyped: true;
   stores: MafStoresList;
   rpcs: MafRpcsList;
 };
diff --git a/packages/react/src/hooks/use-session.ts b/packages/react/src/hooks/use-session.ts
index 2102ce7..bf39a90 100644
--- a/packages/react/src/hooks/use-session.ts
+++ b/packages/react/src/hooks/use-session.ts
@@ -1,19 +1,42 @@
 import { useEffect, useState } from "react";
-import { useMaf } from "../maf-provider";
+import { useMafClient } from "../maf-provider";
 import { SessionInfo } from "@usemaf/client";
-import { MafStatus } from "./use-store";
 
+type UseSessionStatus = "loading" | "ready";
+
+/**
+ * The type returned by the {@link useSession} hook. Represents session loading
+ * state.
+ */
 export type UseSession =
-  | { status: MafStatus.LOADING; data: null }
-  | { status: MafStatus.READY; data: SessionInfo };
+  | { status: "loading"; data: null }
+  | { status: "ready"; data: SessionInfo };
 
+/**
+ * Subscribes to session status and provides session information when ready.
+ *
+ * @example
+ * function App() {
+ *   const session = useSession();
+ *
+ *   // Use discriminated union to handle loading and ready states.
+ *   if (session.status === "loading") {
+ *     return <p>Loading...</p>;
+ *   }
+ *
+ *   // session.status is "ready" here
+ *   return <p>Session User ID: {session.data.id}</p>;
+ * }
+ *
+ * @returns {UseSession} The current session status and data.
+ */
 export function useSession(): UseSession {
-  const [status, setStatus] = useState<MafStatus>(MafStatus.LOADING);
-  const client = useMaf();
+  const [status, setStatus] = useState<UseSessionStatus>("loading");
+  const client = useMafClient();
 
   useEffect(() => {
     const unsubscribe = client.on("ready", () => {
-      setStatus(MafStatus.READY);
+      setStatus("ready");
     });
 
     return () => {
@@ -21,15 +44,15 @@ export function useSession(): UseSession {
     };
   }, [client]);
 
-  if (status === MafStatus.LOADING) {
+  if (status === "loading") {
     return {
-      status: MafStatus.LOADING,
+      status: "loading",
       data: null,
     };
   }
 
   return {
-    status: MafStatus.READY,
+    status: "ready",
     data: client.sessionInfo,
   };
 }
diff --git a/packages/react/src/hooks/use-store.ts b/packages/react/src/hooks/use-store.ts
index 2329753..2844cac 100644
--- a/packages/react/src/hooks/use-store.ts
+++ b/packages/react/src/hooks/use-store.ts
@@ -1,45 +1,95 @@
 import { useEffect, useState } from "react";
-import { useMaf } from "../maf-provider";
+import { useMafClient } from "../maf-provider";
 import type { StoreKeys, StoreSelect } from "@usemaf/client";
 
-export enum MafStatus {
-  LOADING = "loading",
-  READY = "ready",
-}
+type StoreStatus = "loading" | "ready";
 
 type UseStoreDiscUnion<TData, TFallback> =
   | {
-      status: MafStatus.LOADING;
+      status: "loading";
       data: TFallback;
     }
   | {
-      status: MafStatus.READY;
+      status: "ready";
       data: TData;
     };
 
+/**
+ * Subscribes to a *typed* store and reactively provides its data when ready.
+ *
+ * *The data will be be `undefined` until the store is ready.*
+ *
+ * @param storeName The name of the store to subscribe to. This must be
+ * a valid store name defined in the generated types.
+ */
 export function useStore<TName extends StoreKeys>(
   storeName: TName
 ): UseStoreDiscUnion<StoreSelect<TName>, never>;
 
+/**
+ * Subscribes to a *typed* store and reactively provides its data when ready.
+ *
+ * *The data will be be a specified fallback value until the store is ready.*
+ *
+ * @param storeName The name of the store to subscribe to. This must be
+ * a valid store name defined in the generated types.
+ * @param fallback The fallback data to use while the store is loading.
+ */
 export function useStore<TName extends StoreKeys>(
   storeName: string,
   fallback?: StoreSelect<TName>
 ): UseStoreDiscUnion<StoreSelect<TName>, StoreSelect<TName>>;
 
+/**
+ * Subscribes to a *typed* store and reactively provides its data when ready.
+ *
+ * *The data will be be a specified fallback value until the store is ready.*
+ *
+ * @param storeName The name of the store to subscribe to. This must be
+ * a valid store name defined in the generated types.
+ * @param fallback The fallback data to use while the store is loading. *The
+ * fallback data is different from the store data type.*
+ */
 export function useStore<TName extends StoreKeys, TFallback>(
   storeName: TName,
   fallback?: TFallback
 ): UseStoreDiscUnion<StoreSelect<TName>, TFallback>;
 
+/**
+ * Subscribes to a store and reactively provides its data when ready. Use the
+ * type parameter `TData` to specify the expected data shape.
+ *
+ * *The data will be be `undefined` until the store is ready.*
+ *
+ * @param storeName The name of the store to subscribe to.
+ */
 export function useStore<TData>(
   storeName: string
 ): UseStoreDiscUnion<TData, never>;
 
+/**
+ * Subscribes to a store and reactively provides its data when ready. Use the
+ * type parameter `TData` to specify the expected data shape.
+ *
+ * *The data will be be a specified fallback value until the store is ready.*
+ *
+ * @param storeName The name of the store to subscribe to.
+ * @param fallback The fallback data to use while the store is loading.
+ */
 export function useStore<TData, TFallback extends TData>(
   storeName: string,
   fallback: TFallback
 ): UseStoreDiscUnion<TData, TFallback>;
 
+/**
+ * Subscribes to a store and reactively provides its data when ready. Use the
+ * type parameter `TData` to specify the expected data shape.
+ *
+ * *The data will be be a specified fallback value until the store is ready.*
+ *
+ * @param storeName The name of the store to subscribe to.
+ * @param fallback The fallback data to use while the store is loading.
+ */
 export function useStore<TData>(
   storeName: string,
   fallback?: TData
@@ -52,31 +102,31 @@ export function useStore<TData, TFallback>(
   TData | undefined,
   TFallback extends undefined ? never : TFallback
 > {
-  const [status, setStatus] = useState<MafStatus>(MafStatus.LOADING);
+  const [status, setStatus] = useState<StoreStatus>("loading");
   const [data, setData] = useState<TData | TFallback>(
     fallback as TFallback extends undefined ? never : TFallback
   );
-  const client = useMaf();
+  const client = useMafClient();
 
   useEffect(() => {
     const store = client.store<TData>(storeName);
 
     store.init.then(() => {
       setData(store.data);
-      setStatus(MafStatus.READY);
+      setStatus("ready");
       store.on("change", () => setData(store.data));
     });
   }, [client]);
 
-  if (status === MafStatus.LOADING) {
+  if (status === "loading") {
     return {
-      status: MafStatus.LOADING,
+      status: "loading",
       data: fallback as TFallback extends undefined ? never : TFallback,
     };
   }
 
   return {
-    status: MafStatus.READY,
+    status: "ready",
     data: data as TData,
   };
 }
diff --git a/packages/react/src/index.ts b/packages/react/src/index.ts
index 33feef8..051e17d 100644
--- a/packages/react/src/index.ts
+++ b/packages/react/src/index.ts
@@ -1,5 +1,7 @@
-import { MafProvider, type MafProviderProps } from "./maf-provider";
-import { useStore } from "./hooks/use-store";
-import { useSession } from "./hooks/use-session";
-
-export { MafProvider, MafProviderProps, useStore, useSession };
+export {
+  useMafClient,
+  MafProvider,
+  type MafProviderProps,
+} from "./maf-provider";
+export { useStore } from "./hooks/use-store";
+export { useSession } from "./hooks/use-session";
diff --git a/packages/react/src/maf-provider.tsx b/packages/react/src/maf-provider.tsx
index f42acde..b293578 100644
--- a/packages/react/src/maf-provider.tsx
+++ b/packages/react/src/maf-provider.tsx
@@ -1,8 +1,23 @@
-import { ConnectOptions, MafClient, MafServerOptions } from "@usemaf/client";
+import {
+  ConnectOptions,
+  MafClient,
+  MafServerOptions,
+  MafTypes,
+  TypedMafClient,
+} from "@usemaf/client";
 import React, { createContext, useEffect, useState } from "react";
 
 export interface MafProviderProps {
+  /**
+   * What server to connect to.
+   * @see MafServerOptions
+   */
   server: MafServerOptions;
+  /**
+   * Options for connecting to the server. You will need to use this to:
+   * - Include authentication credentials.
+   * - Specify which room to connect to.
+   */
   connectOptions?: ConnectOptions;
 }
 
@@ -12,12 +27,35 @@ export interface MafContextType {
 
 export const MafContext = createContext<MafContextType | null>(null);
 
-export const useMaf = () => {
+/**
+ * Hook to access the `MafClient` instance from context.
+ *
+ * @example
+ * function CounterButton() {
+ *   const maf = useMafClient();
+ *
+ *   return (
+ *     <button onClick={() => maf.rpc("increment_counter", 1)}>
+ *       Increase the counter!
+ *     </button>
+ *   );
+ * }
+ *
+ * @returns An instance of {@link MafClient}, provided by the nearest
+ * {@link MafProvider}.
+ */
+export function useMafClient(): MafTypes["generated"]["__isTyped"] extends true
+  ? TypedMafClient
+  : MafClient {
   const context = React.useContext(MafContext);
   if (!context) throw new Error("useMaf must be used within a MafProvider");
   return context.client;
-};
+}
 
+/**
+ * Provider component that initializes and provides a {@link MafClient} instance
+ * to its child components via context.
+ */
 export const MafProvider: React.FC<
   React.PropsWithChildren<MafProviderProps>
 > = ({ server, connectOptions = { type: "default" }, children }) => {
